pkg/typesystem/typesystem.go

```diff
@@ -862,12 +865,19 @@ func (t *TypeSystem) validateRelation(typeName, relationName string, relationMap
 
 	visitedRelations := map[string]map[string]struct{}{}
 
-	hasEntrypoints, err := hasEntrypoints(t.relations, typeName, relationName, rewrite, visitedRelations)
+	hasEntrypoints, loop, err := hasEntrypoints(t.relations, typeName, relationName, rewrite, visitedRelations)
 	if err != nil {
 		return err
 	}
 
 	if !hasEntrypoints {
+		if loop {
+			return &InvalidRelationError{
+				ObjectType: typeName,
+				Relation:   relationName,
+				Cause:      fmt.Errorf("relation '%s#%s': %w", typeName, relationName, ErrNoEntryPointsLoop),
+			}
+		}
```

Comment:
```suggestion
cause := ErrNoEntrypoints
if loop {
       cause = ErrNoEntryPointsLoop
}
return &InvalidRelationError{
	ObjectType: typeName,
	Relation:   relationName,
	Cause:      cause,
}
```

Reply:
We need to as the error returned from hasEntrypoints usually indicates some lookup error rather than no entry point.  We probably want to reformat the whole thing..but not now.